src/plugins/ruby/ruby/shellvars.rb

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env ruby
+
+require 'kdb'
+
+def array_index_name(name, index)
+  index_str = index.to_s
+  (1..(index.to_s.size - 1)).each { index_str = "_#{index_str}" }
+  "#{name}/##{index_str}"
+end
+
+def key_add_comments(key, comments)
+  comments.each_with_index do |comment_line, comment_index|
+    meta_name = array_index_name 'comment', comment_index
+    if comment_line =~ /^(\s*#\s*)(.*)$/
+      key["#{meta_name}/start"] = $1
+      key[meta_name] = $2
+    else
+      key[meta_name] = comment_line
+      key["#{meta_name}/start"] = ''
+    end
+  end
+end
+
+class FileElement
+
+  def self.create_from_key(key)
+    obj = nil
+    if key['internal/simpleline'] == '1'
+      obj = Simpleline.new
+    else
+      obj = Shellvar.new
+      obj.value = key.value
+    end
+    obj.name = key.basename
+    obj.order = key['order']
+    obj.line = key['line']
+    key.meta.each do |mk|
+      next unless mk.name =~ /^comment\/#_*\d+$/
+
+      comment = ""
+      comment_start_key = "#{mk.name}/start"
+      if key.has_meta? comment_start_key
+        comment += key[comment_start_key]
+      else
+        comment += "# "
+      end
+      comment += mk.value
+      obj.comments << comment
+    end
+
+    return obj
+  end
+
+end
+
+class Simpleline < FileElement
+
+  attr_accessor :name
+  attr_accessor :order
+  attr_accessor :comments
+  attr_accessor :line
+
+  def initialize()
+    @comments = []
+    @order = '@new'
+  end
+
+  def order=(value)
+    if value.is_a? Fixnum
```

Comment:
This line and line 183 produce the following warning:

> warning: constant ::Fixnum is deprecated 

. Looks like [`Integer`](https://bugs.ruby-lang.org/issues/12005) is the correct replacement class. I think we should fix this issue before merging, because the warning is quite annoying. The error message shows up every time we invoke the plugin:

```
» kdb file user/shell/
src/plugins/ruby/ruby/shellvars.rb:69: warning: constant ::Fixnum is deprecated
~/.config/shell.txt
» kdb file user/shell/ | xargs mate
src/plugins/ruby/ruby/shellvars.rb:69: warning: constant ::Fixnum is deprecated
» kdb setmeta ussrc/plugins/ruby/ruby/shellvars.rb:69: warning: constant ::Fixnum is deprecated
» kdb setmeta user/shsrc/plugins/ruby/ruby/shellvars.rb:69: warning: constant ::Fixnum is deprecated
» kdb setmeta user/shell/src/plugins/ruby/ruby/shellvars.rb:69: warning: constant ::Fixnum is deprecated
» kdb setmeta user/shell/key comment "hi key"
src/plugins/ruby/ruby/shellvars.rb:69: warning: constant ::Fixnum is deprecated
src/plugins/ruby/ruby/shellvars.rb:183: warning: constant ::Fixnum is deprecated
```
. As you can see above even if we invoke `kdb complete` Ruby reminds us that we should fix this problem.

Reply:
Thanks for this hint, wasn't aware that Fixnum will be gone by 2.4